core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -620,6 +642,21 @@ class KafkaApis(val requestChannel: RequestChannel,
             topicPartition,
             status.error.exceptionName))
         }
+
+        if (request.header.apiVersion >= 10) {
+          status.currentLeader = {
+            status.error match {
+              case Errors.NOT_LEADER_OR_FOLLOWER | Errors.FENCED_LEADER_EPOCH =>
```

Comment:
produce requests should never receive FENCED_LEADER_EPOCH.

also, shouldn't this go in the above `if` block?

Reply:
will move this into the if block.

why can't produce receive FENCED_LEADER_EPOCH?

Reply:
I think the error is only returned on fetch requests

Reply:
ok, removed FENCED_LEADER_EPOCH from the produce path

Reply:
produce requests do not include a leader epoch => they can never get fenced leader epoch.